internal/core/metadata/controller/http/device_test.go

```diff
@@ -217,13 +185,39 @@ func TestAddDevice(t *testing.T) {
 		{"Invalid - no protocols", []requests.AddDeviceRequest{noProtocols}, http.StatusBadRequest},
 		{"Invalid - empty protocols", []requests.AddDeviceRequest{emptyProtocols}, http.StatusBadRequest},
 		{"Invalid - invalid protocols", []requests.AddDeviceRequest{invalidProtocols}, http.StatusInternalServerError},
-		{"Valid - device service unavailable", []requests.AddDeviceRequest{serviceUnavailable}, http.StatusCreated},
+		{"Invalid - device service unavailable", []requests.AddDeviceRequest{serviceUnavailable}, http.StatusServiceUnavailable},
 	}
 	for _, testCase := range tests {
 		t.Run(testCase.name, func(t *testing.T) {
 			jsonData, err := json.Marshal(testCase.request)
 			require.NoError(t, err)
 
+			var responseEnvelope types.MessageEnvelope
+			mockMessaging := &messagingMocks.MessageClient{}
+			mockMessaging.On("Publish", mock.Anything, mock.Anything).Return(nil)
```

Comment:
Why is mock on Publish needed here? Request encapsulates it, should mocking Request will keep it from actually being called.

Reply:
This is called in valid case for publishing system event.
I added ffb0bc6a7dd4eab0367d8b6f6b6d530a5be87bbf to better test this behavior